htmresearch/frameworks/clustering/sdr_clustering.py

```diff
@@ -0,0 +1,213 @@
+import logging
+import numpy as np
+
+from htmresearch.frameworks.clustering.distances import clusterDist
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.DEBUG)
+
+
+
+class Point(object):
+  def __init__(self, pointId, value, label):
+    self._id = pointId
+    self._value = value
+    self._label = label
+
+
+  def getId(self):
+    return self._id
+
+
+  def setLabel(self, label):
+    self._label = label
+
+
+  def getLabel(self):
+    return self._label
+
+
+  def setValue(self, value):
+    self._value = value
+
+
+  def getValue(self):
+    return self._value
+
+
+
+class Cluster(object):
+  def __init__(self, id, points=None, label=None):
+    self._id = id
+    self._label = label
+    if points:
+      self._points = points
+    else:
+      self._points = []
+    _LOGGER.debug("New cluster %s created!" % self._id)
+
+
+  def add(self, point):
+    self._points.append(point)
+
+
+  def setLabel(self, name):
+    self._label = name
+
+
+  def getLabel(self):
+    return self._label
+
+
+  def getId(self):
+    return self._id
+
+
+  def setPoints(self, points):
+    self._points = points
+
+
+  def getPoints(self):
+    return self._points
+
+
+  def size(self):
+    return len(self._points)
+
+
+  def merge(self, cluster):
+    for point in cluster.getPoints():
+      self.add(point)
+
+
+
+class Clustering(object):
+  def __init__(self,
+               mergeThreshold,
+               anomalousThreshold,
+               stableThreshold,
+               fistClusterId=0):
+
+    # Clusters
+    self.newCluster = Cluster(fistClusterId)
+    self.clusters = []
+
+    # Anomaly Score Thresholds
+    self.anomalousThreshold = anomalousThreshold  # to create new cluster
+    self.stableThreshold = stableThreshold  # to add point to cluster
+
+    # Cluster distance threshold
+    self.mergeThreshold = mergeThreshold
+
+
+  def findClosestClusters(self, cluster):
+    """
+    
+    :param cluster: 
+    :return: ordered list of clusters and their distances 
+    """
+    dists = []
+    for c in self.clusters:
+      d = clusterDist(c.getPoints(), cluster.getPoints())
+      dists.append((d, c))
+
+    return sorted(dists)
+
+
+  def addCluster(self, cluster):
```

Comment:
Right now there is no need for a remove cluster function. We only merge clusters.